### PR TITLE
 Single Page Application (SPA)

### PR DESCRIPTION
**Pull request by ghifo**

### CODE REVIEW

File: homework/index.js [hunk 1]
```diff
@@ -29,19 +29,149 @@
     });
     return elem;
   }
+  // ==========start===========
+  /* cSpell: disable */
+  const repoInfo = (index, repoName) => {
```
Comment (eidosam) [inline homework/index.js]: You could send the repo data instead of passing both array and index.

File: homework/index.js [hunk 2]
```diff
@@ -29,19 +29,149 @@
...
+    const container = createAndAppend('div', document.getElementById('root'), {
+      class: 'container',
+      id: 'container',
+    });
+
+    // left div
+    const leftDiv = createAndAppend('div', container, {
```
Comment (eidosam) [inline homework/index.js]: You could name it `repoContainer` instead?

File: homework/index.js [hunk 3]
```diff
@@ -29,19 +29,149 @@
     });
     return elem;
   }
+  // ==========start===========
+  /* cSpell: disable */
```
Comment (eidosam) [inline homework/index.js]: Please enable **Spell Checker** and fix spelling errors if there is any.

**Comment by wouterkleijn** [comment]
Dear @eidosam Thanks for your comments. @ghifo would be great if you could reply to some of the comments and make the necessary improvements - thanks!

**Comment by ghifo** [comment]
Sure!

**Comment by ghifo** [comment]
I know I ought to ask that before,


@eidosam <https://github.com/eidosam>


I just wanted to discuss the comments today if possible!

File: homework/index.js (hunk 1 above)
Comment (ghifo) [reply homework/index.js]: I tried to do so but I got 'Network request failed'

File: homework/index.js (hunk 2 above)
Comment (ghifo) [reply homework/index.js]: OK

File: homework/index.js (hunk 3 above)
Comment (ghifo) [reply homework/index.js]: I did it at the end of the code, I used naming which are considered as spell error. I will avoid that next time.

**Pull request closed without merging**